### PR TITLE
LRSD-3260 [CP 2.0] Allow multiple roles in the invite flow and in the team members table

### PR DESCRIPTION
**Pull request by pedroliveirav**
[LRSD-3260](https://liferay.atlassian.net/browse/LRSD-3260)

### CODE REVIEW

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.jsx [hunk 1]
```diff
@@ -412,15 +423,36 @@ const InviteTeamMembersPage = ({
...
+													`invites[${index}].role[${roleIndex}]`,
+													accountRoles?.find(
+														({ id }) => id === +updateRole.value
+													)
 												)
-											)
-										}
+											})
+										}}
+										setRoleSelectorFilled={setRoleSelectorFilled}
```
Comment (amosfong) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.jsx:455]: missing semicolon

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/RoleSelectorDropdown/index.jsx [hunk 2]
```diff
@@ -0,0 +1,218 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
```
Comment (amosfong) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/RoleSelectorDropdown/index.jsx:2]: Could you go through this file again and check the formatting? There's some extra spaces, missing semicolons, etc.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/TeamMembersTable/components/columns/RolesColumn/components/RolesDropdown/RolesDropdown.jsx [hunk 3]
```diff
@@ -18,84 +21,45 @@ const RolesDropdown = ({
...
-				>
-					<div className="text-truncate">
-						{i18n.translate(getKebabCase(selectedAccountRoleName))
-							? i18n.translate(
-									getKebabCase(selectedAccountRoleName)
-							  )
-							: selectedAccountRoleName}
-					</div>
+			return previousItem;
+		}, { partnerMemberRoles: { active: undefined, roles: [] } }))
```
Comment (amosfong) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/TeamMembersTable/components/columns/RolesColumn/components/RolesDropdown/RolesDropdown.jsx:52]: I think this line should be expanded into new lines

**Comment by pedroliveirav** [comment]
> Did you run gw formatsource? There seems to be some extra whitespace that I think it would automatically remove.

Yeah, I ran gw formatSource, but it didn't adjust this. I've corrected the formatting of the code now, could you check again?

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/TeamMembersTable/components/columns/RolesColumn/components/RolesDropdown/RolesDropdown.jsx (hunk 3 above)
Comment (pedroliveirav) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/TeamMembersTable/components/columns/RolesColumn/components/RolesDropdown/RolesDropdown.jsx]: Fixed code formatting.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/RoleSelectorDropdown/index.jsx (hunk 2 above)
Comment (pedroliveirav) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/RoleSelectorDropdown/index.jsx:2]: Done.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/styles/containers/setup-forms/_invite-team-members-form.scss [hunk 4]
```diff
@@ -14,3 +14,24 @@
 .cp-badge-error-message {
 	width: 250px;
 }
+
+.role-selector-container {
```
[the remark this answers is not in the document]
Comment (pedroliveirav) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/styles/containers/setup-forms/_invite-team-members-form.scss:18]: Fixed.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.jsx (hunk 1 above)
Comment (pedroliveirav) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.jsx]: Code formatting fixed.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/TeamMemberInputs/index.jsx [hunk 5]
```diff
@@ -3,24 +3,30 @@
  * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
  */
 
-import {ClayInput} from '@clayui/form';
+import { ClayInput } from '@clayui/form';
 import ClayIcon from '@clayui/icon';
-import {useEffect, useMemo} from 'react';
-import {useAppPropertiesContext} from '~/common/contexts/AppPropertiesContext';
+import { useEffect, useMemo, useState } from 'react';
```
[the remark this answers is not in the document]
Comment (pedroliveirav) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/TeamMemberInputs/index.jsx]: Code formatting fixed.

**Comment by amosfong** [comment]
Hi @pedroliveirav , i made some SF changes.
https://github.com/amosfong/liferay-portal/commit/bd420ae9ab0e6caf351d51c3a6f4ca452d3fccd0

I was testing locally and I ran into a couple issues:
- partner member always shows up. It should only show up for accounts with partner subscriptions. See getRolesFiltered()
- when I click partner member there should be a popup right? Do I need to do anything to get this to show up? It doesn't work on my local.

**Comment by samuelleolopes** [comment]
> Hi @pedroliveirav , i made some SF changes. [amosfong@bd420ae](https://github.com/amosfong/liferay-portal/commit/bd420ae9ab0e6caf351d51c3a6f4ca452d3fccd0)
> 
> I was testing locally and I ran into a couple issues:
> 
> * partner member always shows up. It should only show up for accounts with partner subscriptions. See getRolesFiltered()
> * when I click partner member there should be a popup right? Do I need to do anything to get this to show up? It doesn't work on my local.

Hi Amos, on my local the styles/css of the dropdown menu are fine.

image

**Pull request closed without merging**